tests,rpc: update for BadLogLines test failures

jcsp opened the pull request:
## Cover letter

- RpkConfigTest needed an allow list when restarting nodes
- Rare misleading shutdown errors from `simple_protocol::dispatch_method_once` are silenced

## Release notes

* none

dotnwat commented:
ahh looks like a little conflict. I'd fix it in the UI but that seems to be problematic for the CI rebase.

jcsp commented:
Updated to fix merge conflict, and added a commit for MetadataTest's allow-list from a failure last night (https://buildkite.com/vectorized/redpanda/builds/6582#fb73ccea-452f-4716-82ca-5d6548430bb2)
